stl/inc/flat_set

```diff
@@ -768,17 +764,19 @@ public:
 _EXPORT_STD template <class _Kty, class _Keylt, class _Container, class _Pred>
 _Container::size_type erase_if(flat_set<_Kty, _Keylt, _Container>& _Val, _Pred _Predicate) {
     // clears the container to maintain the invariants when an exception is thrown (N4950 [flat.set.erasure]/5)
-    _Clear_guard<flat_set<_Kty, _Keylt, _Container>> _Guard{_STD addressof(_Val)};
-    const auto _Erased_count = _STD _Erase_remove_if(_Val, _STD _Pass_fn(_Predicate));
+    _Container& _Cont = const_cast<_Container&>(_Val._Get_container());

```

Comment:
Why `const_cast` every use of `_Get_container()` instead of making it non-`const`?

Reply:
yes this looks awkward... I thought if `flat_set` want to expose its container, it should always be const to avoid outside modification... `erase_if` will be only exception, but it's unfortunately also the only user of this function now... Is it ok to remove `_Get_container` and befriend `erase_if` instead?

Reply:
Is it allowed to define erase_if friend in the class?

Reply:
I believe there are observable differences between hidden friends and namespace-scope functions, so we can't make such a transformation.

Reply:
I've made `_Get_container` non-const, but renamed it to `_Get_container_for_erase_if`, as I'm concerned about exposing the non-const container under a too general name.